src/lib/utils/alignment_buffer.h

```diff
@@ -0,0 +1,219 @@
+/*
+ * Alignment buffer helper
+ * (C) 2023 Jack Lloyd
+ *     2023 René Meusel - Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ */
+
+#ifndef BOTAN_ALIGNMENT_BUFFER_H_
+#define BOTAN_ALIGNMENT_BUFFER_H_
+
+#include <botan/concepts.h>
+#include <botan/internal/stl_util.h>
+
+#include <array>
+#include <optional>
+#include <span>
+
+namespace Botan {
+
+enum class AlignmentBufferFinalBlock : size_t {
+   IS_NOT_SPECIAL = 0,
+   MUST_BE_DEFERRED = 1,
```

Comment:
IMO `enum class` members should not be `SHOUTING_CASE`

Reply:
Also for future users, briefly document the usage of this enum in a comment.

Reply:
Thanks for the approval. I just pushed another change (had forgotten to add documentation of the enum class).